tests/dialects/test_clickhouse.py

```diff
@@ -417,6 +418,15 @@ def test_cte(self):
         self.assertIsInstance(query.args["with"].expressions[0].this, exp.Subquery)
         self.assertEqual(query.args["with"].expressions[0].alias, "y")
 
+        query = "WITH 1 AS var SELECT var"
+        for error_level in [ErrorLevel.IGNORE, ErrorLevel.RAISE, ErrorLevel.IMMEDIATE]:
+            self.assertEqual(
+                parse_one(query, dialect=self.dialect, error_level=error_level).sql(
+                    dialect=self.dialect
+                ),
+                query,
```

Comment:
```suggestion
```suggestion
                self.parse_one(query, error_level=error_level).sql(dialect=self.dialect),
                query,
```
```

Reply:
If the read dialect isn't set as `Clickhouse`, the expression-based CTE isn't parseable so this unit test would fail if I'm not losing something

Reply:
Note the `self.` in front of `parse_one` - in tests you can do this and the dialect will be set for you automatically.

Reply:
Required a small refactor in `Validator::parse_one` to allow for kwarg propagation